src/platform_linux/laio.c

```diff
@@ -378,34 +518,58 @@ laio_cleanup(io_handle *ioh, uint64 count)
    uint64          i;
    int             status;
 
+   threadid tid = platform_get_tid();
+
    io = (laio_handle *)ioh;
 
    // Check for completion of up to 'count' events, one event at a time.
    // Or, check for all outstanding events (count == 0)
    for (i = 0; ((count == 0) || (i < count)); i++) {
-      status = io_getevents(io->ctx, 0, 1, &event, NULL);
+      status = io_getevents(io->ctx[tid], 0, 1, &event, NULL);
```

Comment:
Each thread will now poll for the completion of its IO events.

@ajhconway -- it's subtle stuff like this that will need close examination. Will be good if you can dwell into this thread-specific IO context re-design, to see if you can spot any gaping holdes.